src/login/getAuthLibrary.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { callWithTelemetryAndErrorHandlingSync, IActionContext } from "vscode-azureextensionui";
+import { AuthLibrary, authLibrarySetting } from "../constants";
+import { ext } from "../extensionVariables";
+import { getSettingValue } from "../utils/settingUtils";
+
+export function getAuthLibrary(): AuthLibrary {
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	return (callWithTelemetryAndErrorHandlingSync('getAuthLibrary', (context: IActionContext) => {
```

Comment:
I don't know for sure without trying, but this might eliminate the need for the non-null assertion (and consequently the eslint suppression, and one pair of parentheses):
```suggestion
	return (callWithTelemetryAndErrorHandlingSync('getAuthLibrary', (context: IActionContext): AuthLibrary => {
```

Otherwise, `as AuthLibrary` below might work too.